### ISSUE TITLE
Allow to suppress MISSING_ENV_FILE when using own convention with optional paths

### ISSUE THREAD

**Issue by thdk** [dotenvx/dotenvx#483]
I see that no `MISSING_ENV_FILE` error is logged when using the `nextjs` convention. 

https://github.com/dotenvx/dotenvx/blob/main/src/cli/actions/run.js#L65-L71

It would be nice to be allowed to suppress this error as well when making a custom convention based on multiple paths where all paths are optional.

```typescript
const mode: 'staging' | 'acceptance' | 'production' = process.env.MODE;

dotenvx.config({
  path: [`./.env.${mode}.local`, `./.env.${mode}`, '.env.local', '.env'],
});
```

This gives the following error output:

```sh
[MISSING_ENV_FILE] missing .env.local file (/Users/thdk/repos/dotenvx-demo/.env.local)
[MISSING_ENV_FILE] ? add one with [echo "HELLO=World" > .env.local]
```

Something like this?

```sh
dotenvx.config({
  path: [`./.env.${mode}.local`, `./.env.${mode}`, '.env.local', '.env'],
  allowMissingEnvFiles: true,
});
```
Implementation could be done without breaking current logic:
```typescript
if (error.code === 'MISSING_ENV_FILE') {
      if (options.allowMissingEnvFiles === undefined ? !options.convention :  !options.allowMissingEnvFiles ) {
        console.error(error.message)
        if (error.help) {
          console.error(`${error.help} and re-run [dotenvx run -- ${commandArgs.join(' ')}]`)
        }
 }
```

**Comment by motdotla** [comment]
this is a good point you bring up. let me think on it a bit but your suggestion or something close sounds good

**Comment by motdotla** [comment]
i'm leaning toward a more generic `--omit`, `--surpress`, `--mute`, or similar where you can mute specific errors.

```
dotenvx run --mute=MISSING_ENV_FILE
```

```
dotenvx.config({
  path: [`./.env.${mode}.local`, `./.env.${mode}`, '.env.local', '.env'],
  mute: ['MISSING_ENV_FILE']
})
```

**Comment by thdk** [comment]
First of all thanks for taking this into consideration.

I think the ideal solution would allow to indicate on env file level which
are required and which are optional.

Like:

```js
paths: [{ optional: true, file: '.env'}, { file: '.env.foo' }]
```

**Comment by motdotla** [comment]
yeah, i understand the thought there but that breaks a historical pattern that I don't want to break. too much muscle memory there for devs around the world.

furthermore, with the addition of the encrypted `.env*` files it will become increasingly rare users load multiple .env files at once. in fact, it was an anti-pattern I ecommended against, but Next.js popularized it. 

(to be fair I never provided an alternative and so .env.local filled the gap of configuring local db config. but then again this was because many next.js users were committing their .env file - also an anti-pattern. if they were just sharing a .env.example and then configuring it locally there would have never been a need for .env.local.)

so in summary, I don't want to go down this path of optional paths. rather for those that really don't want the stderr they can suppress it (as well as any other errors).

**Comment by motdotla** [comment]
this is going out as `1.29.0` now.

<img width="2158" alt="image" src="https://github.com/user-attachments/assets/5817cf65-8edd-44cf-a484-a970c59d5b27">

<img width="1782" alt="image" src="https://github.com/user-attachments/assets/5deccfe2-679f-45ba-869c-61352fdc13fa">

**Comment by thdk** [comment]
@motdotla this works in the latest version (thanks a lot) but types haven't been updated.

If you allow me some time, I can submit a PR with that. Just wanted to let you know already.

**Comment by motdotla** [comment]
that would be much appreciated. thanks @thdk.

(someone needs to make a /tip command here in github. i'd love to be able to doll out tips for this kind of help)

**Comment by thdk** [comment]
Someone already fixed it 🚀 

https://github.com/dotenvx/dotenvx/commit/1b7e2e163d8e5b4c9f0483d535c8f5bd70019fe0
